File: BrickABracket.Models/Base/Tournament.cs
using System.Collections.Generic;
using BrickABracket.Models.Interfaces;

namespace BrickABracket.Models.Base
{
    public class Tournament : TournamentSummary, IDBItem
    {
        public List<Category> Categories { get; set; } = new List<Category>();
        public List<int> MocIds { get; set; } = new List<int>();
        // Inherited properties:
        // public string TournamentType { get; set; }
        // public string Name { get; set; }
        // public int _id { get; set; }
    }
}

File: BrickABracket/ClientApp/src/app/core/tournaments/tournament.ts
import {Category} from './category';

export class Tournament {
    categories: Array<Category>;
    mocIds: Array<number>;
    tournamentType: string;
    name: string;
    _id: number;
}
export interface TournamentSummary {
    tournamentType: string;
    name: string;
    _id: number;
}
export interface TournamentMetadata {
    tournament: Tournament;
    categoryIndex: number;
    roundIndex: number;
    matchIndex: number;
}
